core/src/workflows/get_link_result.rs

```diff
@@ -56,36 +46,79 @@ pub async fn get_link_result_workflow<'a>(
                                     tag: link_data.link().tag().to_string(),
                                 })
                             },
-                            Ok(None) => {
+                            None => {
                                 Err(HolochainError::ErrorGeneric(
-                                    format!("Could not get link entry for address stored in the EAV entry {}", link),
+                                    format!("Could not get link entry for address stored in the EAV entry {:?}", link),
                                 ))
                             }
-                            Err(e) => {
-                                Err(HolochainError::ErrorGeneric(
-                                    format!("Error retrieveing link: {:?}", e),
-                                ))
-                            },
                             _ => {
                                 Err(HolochainError::ErrorGeneric(
                                     format!("Unknown Error retrieveing link. Most likely EAV entry points to non-link entry type"),
                                 ))
                             }
                         }
                     },
-                ),
             )
-        })
         .partition(Result::is_ok);
 
     if errors.is_empty() {
         Ok(GetLinksResult::new(
             link_results.into_iter().map(|s| s.unwrap()).collect(),
         ))
+    } else {
+        Err(HolochainError::ErrorGeneric(
+            "Could not get links".to_string(),
+        ))
+    }
+}
+
+async fn get_link_caches<'a>(
```

Comment:
I was confused by this name and trying to understand in what way it caches anything. I do understand this function to be necessary exactly because of the performance problem I mentioned above: `get_links` returns addresses of `Entry::LinkAdd` entries, so this function gets those link entries. Should we rename it to something like:
```suggestion
async fn get_link_add_entries<'a>(
```
?